### PR TITLE
[IAM-4740] add docs for the new SSO

### PR DESCRIPTION
**Pull request by przemek-sl**
### Description, Motivation and Context
We are going to release the new SSO implementation. Any new SSO integration is going to be set up using the new flow. I'm adding the documentation for the new SSO ([second commit](https://github.com/saucelabs/sauce-docs/commit/e7a5f9b277a052f4ab324c9175294aca95e07a38)). 

However, we have to keep the existing docs for customers who still use the old SSO, until they migrate to the new SSO. That is why I marked the existing SSO docs as deprecated ([first commit](https://github.com/saucelabs/sauce-docs/commit/7835206eb48b69ebb17a2bc8f39731bee641f02e)).

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist
- [x] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/main/docs/contributing.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: docs/basics/sso/setting-up-sso.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+you set it up. Unlike to the [old SSO implementation](/basics/sso-deprecated/setting-up-single-sign-on) which is already deprecated.
+:::
+
+## Setting Up Identity Provider
+
+1. Obtain SAML metadata from Sauce Labs Service Provider which is served under [this link](https://accounts.saucelabs.com/am/sso/metadata/https%3A%2F%2Faccounts.saucelabs.com%2Fsp).
+2. Log in to your IdP admin account, such as Okta, OneLogin, or Microsoft Azure Active Directory, etc.
+3. Create a new SAML application.
+   - In the IdP, go to the "Applications" or "Integrations" section and create a new SAML application.
+   - Follow the instructions from your IdP to provide the Sauce Labs SAML metadata obtained in step 1. If your IdP does not allow uploading SP metadata, set up the integration manually using the information provided in the paragraphs below.
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:33]: Add a link to the heading for this "in the paragraphs below" 

```suggestion
   - Follow the instructions from your IdP to provide the Sauce Labs SAML metadata obtained in step 1. If your IdP does not allow uploading service provider  metadata, set up the integration manually using the information provided in the paragraphs below.
```

File: docs/basics/sso/setting-up-sso.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+4. Export SAML metadata of your new SAML application.
+   - It will be used to set up the SSO at Sauce Labs in [Team Management Panel](#integrating-with-sauce-labs-service-provider).
+
+### Service Provider SAML Requirements
+
+Here is the list of settings which are required by Sauce Labs Service Provider:
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:39]:
```suggestion
Here is the list of settings that are required by Sauce Labs Service Provider:
```

File: docs/basics/sso/setting-up-sso.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+| Assertion Consumer Service   | `https://accounts.saucelabs.com/am/AuthConsumer/metaAlias/authtree/sp` |
+| Recipient URL                | `https://accounts.saucelabs.com/am/AuthConsumer/metaAlias/authtree/sp` |
+| Destination URL              | `https://accounts.saucelabs.com/am/AuthConsumer/metaAlias/authtree/sp` |
+| Name ID format               | `urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress`               |
+| Authentication Context Class | `urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport`    |
+| Binding                      | `urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST`                       |
+
+### Signature
+
+Either SAML Assertion or SAML response **must be signed**.
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:54]:
```suggestion
Either the SAML Assertion or SAML Response **must be signed**.
```

File: docs/basics/sso/setting-up-sso.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+A digital signature is required to ensure that only your IdP generated the assertion.
+
+### Encryption
+
+Encryption of SAML assertion **is not mandatory**.
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:60]:
```suggestion
Encryption of the SAML Assertion **is not mandatory**.
```

File: docs/basics/sso/setting-up-sso.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+You can enable encryption if you want an additional layer of security. You will need to use the certificate (with alias _encryption_) from [the Sauce Labs metadata](https://accounts.saucelabs.com/am/sso/metadata/https%3A%2F%2Faccounts.saucelabs.com%2Fsp).
+
+:::tip
+
+#### Possible Issues
+
+If you start to see errors after enabling encryption, make sure that you use the encryption certificate which is advertised in Sauce Labs metadata (in the tag `<KeyDescriptor use="encryption">`) and the encryption algorithm matches the algorithm provided in Sauce Labs metadata (value of the attribute `Algorithm` in the tag `EncryptionMethod`).
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:68]:
```suggestion
If you start to see errors after enabling encryption, make sure that you use the encryption certificate that is advertised in Sauce Labs metadata (in the tag `<KeyDescriptor use="encryption">`) and the encryption algorithm matches the algorithm provided in Sauce Labs metadata (value of the attribute `Algorithm` in the tag `EncryptionMethod`).
```

File: docs/basics/sso/setting-up-sso.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+:::
+
+### Name ID
+
+Name ID format must be set to `urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress`.
+
+The value of the attribute NameID in SAML Response must be also **a valid email address**. We do not accept values which are not email addresses, for instance usernames, IDs, etc.
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:75]:
```suggestion
The value of the attribute NameID in SAML Response must be **a valid email address**. We do not accept values which are not email addresses, for instance usernames, IDs, etc.
```

File: docs/basics/sso/setting-up-sso.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+
+### SAML Claims
+
+Sauce Labs Service Provider does not require to include any extra attributes in SAML Response.
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:79]:
```suggestion
Sauce Labs Service Provider does not require any extra attributes in the SAML Response.
```

File: docs/basics/sso/setting-up-sso.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+
+### Certificate rotation
+
+:::note
+This section is relevant only for identity providers which enabled encryption of SAML assertions.
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:84]:
```suggestion
This section is relevant only for identity providers that enabled encryption of SAML Assertions.
```

File: docs/basics/sso/setting-up-sso.md [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+:::
+
+If you enabled encryption of SAML assertions in your IdP, it means that you uploaded the certificate which is served in Sauce Labs [metadata](https://accounts.saucelabs.com/am/sso/metadata/https%3A%2F%2Faccounts.saucelabs.com%2Fsp).
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:87]:
```suggestion
If you enabled encryption of SAML Assertions in your IdP,  you must upload the certificate that is provided in Sauce Labs [metadata](https://accounts.saucelabs.com/am/sso/metadata/https%3A%2F%2Faccounts.saucelabs.com%2Fsp).
```

File: docs/basics/sso/setting-up-sso.md [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+
+The certificate is valid one year. **120 days** before the expiration date the new certificate will appear in Sauce Labs [metadata](https://accounts.saucelabs.com/am/sso/metadata/https%3A%2F%2Faccounts.saucelabs.com%2Fsp). Upload the new certificate in your IdP.
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:89]:
```suggestion
The certificate is valid for one year. **120 days** before the expiration date the new certificate will be available in Sauce Labs [metadata](https://accounts.saucelabs.com/am/sso/metadata/https%3A%2F%2Faccounts.saucelabs.com%2Fsp). Upload the new certificate in your IdP.
```

File: docs/basics/sso/setting-up-sso.md [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+
+If you encounter any issues during the certificate rotation, you can safely go back to the old certificate (until the expiration date). Both certificates will work.
+
+:::warning
+If you do not replace the encryption certificate in your IdP on time, once the certificate expires, any user from your organization will not be able to authenticate via SSO.  
+:::
+
+## Integrating With Sauce Labs Service Provider
+
+1. On Sauce Labs, click **Account** and then click **Team Management**.
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:99]:
```suggestion
1. On Sauce Labs, click **Account**,  and then click **Team Management**.
```

File: docs/basics/sso/setting-up-sso.md [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+
+3. On the **Organization Settings** page, click the **SINGLE SIGN-ON** tab.
+
+<img src={useBaseUrl('img/basics/sso/setup-sso-sp-sso-tab.png')} alt="SSO Tab" width="400" />
+
+4. Upload the SAML metadata file provided by your IdP.
+
+<img src={useBaseUrl('img/basics/sso/setup-sso-sp-upload-metadata.png')} alt="Upload Metadata Button" width="500" />
+
+5. Under **Status** section, toggle the switch to **Enabled**. For more information about adding new users and SSO considerations, see [Adding and Deactivating Users](/basics/acct-team-mgmt/adding-deactivating-users).
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:115]:
```suggestion
5. In the **Status** section, toggle the switch to **Enabled**. For more information about adding new users and SSO considerations, see [Adding and Deactivating Users](/basics/acct-team-mgmt/adding-deactivating-users).
```

File: docs/basics/sso/setting-up-sso.md [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+
+<img src={useBaseUrl('img/basics/sso/setup-sso-enable-sso.png')} alt="Enable SSO" width="500" />
+
+6. [Test the integration](/basics/sso/logging-in-via-sso).
+
+### Email domains
+
+We strongly recommend to assign your company email domains to your Sauce Labs organization. It will allow users who do not have account at Sauce Labs to initiate SSO at [the Sauce Labs login page](https://accounts.saucelabs.com).
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:123]:
```suggestion
We strongly recommend that you  assign your company email domains to your Sauce Labs organization. This allows users who do not have an account at Sauce Labs to initiate SSO at [the Sauce Labs login page](https://accounts.saucelabs.com).
```

File: docs/basics/sso/setting-up-sso.md [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+
+New users can sign up via SP or IdP-initiated SSO and the new accounts at Sauce Labs will be created automatically.
+
+#### Team placement
+
+Users who are created via SSO are placed into the default team.
+
+Organization admins can [assign users to other teams](/basics/acct-team-mgmt/assigning-removing-users-teams).
+
+## Not supported features
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:153]:
```suggestion
##  Unsupported features
```

File: docs/basics/sso/setting-up-sso.md [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+
+The following SSO-related features are currently not supported:
+
+- SCIM (System for Cross-domain Identity Management)
+- SLO (Single Logout)
+- a staging environment for pre-integration testing of SSO
+- integration with multiple identity provider in a single Sauce Labs organization
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:160]:
```suggestion
- integration with multiple identity providers in a single Sauce Labs organization
```

File: docs/basics/sso/logging-in-via-sso.md [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+
+<p><span className="sauceDBlue">Enterprise Plans only</span></p>
+
+When using SAML SSO, we have two methods for starting Single Sign-On (SSO): [SP-initiated](#service-provider-sp-initiated-sso) or [IdP-initiated](#identity-provider-idp-initiated-sso).
+
+## Service Provider (SP) initiated SSO
+
+1. Go to [the Sauce Labs login page](https://accounts.saucelabs.com).
+
+2. Click the button **`SSO`**.
```
Comment (aliciakrebs) [inline docs/basics/sso/logging-in-via-sso.md:17]:
```suggestion
2. Click **SSO**.
```

File: docs/basics/sso/logging-in-via-sso.md [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+
+<img src={useBaseUrl('img/basics/sso/setup-sso-sso-btn.png')} alt="SSO Login Button" width="500" />
+
+3. Input your company email address and click **`Log in`**.
```
Comment (aliciakrebs) [inline docs/basics/sso/logging-in-via-sso.md:21]:
```suggestion
3. Enter your email address and click **Log in**.
```

File: docs/basics/sso/logging-in-via-sso.md [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+
+<img src={useBaseUrl('img/basics/sso/login-sso-input-email.png')} alt="Input SSO Email" width="500" />
+
+4. You will be redirected to your identity provider in order to authenticate with your company credentials.
```
Comment (aliciakrebs) [inline docs/basics/sso/logging-in-via-sso.md:25]:
```suggestion
4. You will be redirected to your identity provider to authenticate with your company credentials.
```

File: docs/basics/sso/logging-in-via-sso.md [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+
+5. If you are already authenticated at your identity provider, you will be redirected to Sauce Labs dashboard.
```
Comment (aliciakrebs) [inline docs/basics/sso/logging-in-via-sso.md:27]:
```suggestion
5. If you are already authenticated at your identity provider, you will be redirected to the Sauce Labs dashboard.
```

File: docs/basics/sso/logging-in-via-sso.md [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+
+:::tip
+If new users are not able to log in via SP-initiated flow, most probably your company email domains were not assigned to your organization. Check [here](/basics/sso/setting-up-sso#email-domains) how to do that.
```
Comment (aliciakrebs) [inline docs/basics/sso/logging-in-via-sso.md:30]:
```suggestion
If new users are not able to log in via the SP-initiated flow, make sure that  your [company email domains were assigned to your organization](/basics/sso/setting-up-sso#email-domains).
```

File: docs/basics/sso/logging-in-via-sso.md [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+:::
+
+### Deep linking
+
+Copying and pasting links is the quickest way to share pages.
+
+If you paste a link to for example a test results page at Sauce Labs and you are not authenticated. You will be redirected to the Sauce Labs login page.
```
Comment (aliciakrebs) [inline docs/basics/sso/logging-in-via-sso.md:37]:
```suggestion
If you paste a link to, for example,  a test results page at Sauce Labs and you are not authenticated. You will be redirected to the Sauce Labs login page.
```

File: docs/basics/sso/logging-in-via-sso.md [hunk 22]
```diff
@@ -0,0 +1,57 @@
...
+
+<img src={useBaseUrl('img/basics/sso/login-sso-login-page.png')} alt="Login Page" width="500" />
+
+2. Follow the steps from [the SP-initiated flow](#service-provider-sp-initiated-sso):
+
+   - Click the button **`SSO`**.
```
Comment (aliciakrebs) [inline docs/basics/sso/logging-in-via-sso.md:43]:
```suggestion
   - Click **SSO**.
```

File: docs/basics/sso/logging-in-via-sso.md [hunk 23]
```diff
@@ -0,0 +1,57 @@
...
+   - Input your company email address.
```
Comment (aliciakrebs) [inline docs/basics/sso/logging-in-via-sso.md:44]:
```suggestion
   - Enter your company email address.
```

File: docs/basics/sso/logging-in-via-sso.md [hunk 24]
```diff
@@ -0,0 +1,57 @@
...
+   - Click the button **`Log in`**.
```
Comment (aliciakrebs) [inline docs/basics/sso/logging-in-via-sso.md:45]:
```suggestion
   - Click **Log in**.
```

File: docs/basics/sso/logging-in-via-sso.md [hunk 25]
```diff
@@ -0,0 +1,57 @@
...
+
+3. And you will be redirected to the requested page.
+
+## Identity Provider (IdP) initiated SSO
+
+1. Log into your identity provider dashboard, for example Okta.
+
+2. Click on **`Sauce Labs`** tile in your applications catalogue.
```
Comment (aliciakrebs) [inline docs/basics/sso/logging-in-via-sso.md:53]:
```suggestion
2. Click the **Sauce Labs** tile in your applications catalog.
```

File: docs/basics/sso/setting-up-sso.md [hunk 26]
```diff
@@ -0,0 +1,160 @@
...
+- An enterprise license for Sauce Labs
+- Organization admin access (see [User Roles](/basics/acct-team-mgmt/managing-user-info) for more information)
+
+## Setup Overview
+
+First, set up a new SAML application in your [Identity Provider (IdP)](#setting-up-identity-provider). Next, enable and configure SSO in [Sauce Labs Team Management Panel](#integrating-with-sauce-labs-service-provider).
+
+:::note
+SSO at Sauce Labs is global. In other words, it does not matter in which region 
+you set it up. Unlike to the [old SSO implementation](/basics/sso-deprecated/setting-up-single-sign-on) which is already deprecated.
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:24]:
```suggestion
you set it up. Unlike to the [old SSO implementation](/basics/sso-deprecated/setting-up-single-sign-on), which has been deprecated.
```

File: docs/basics/sso/setting-up-sso.md [hunk 27]
```diff
@@ -0,0 +1,160 @@
...
+:::
+
+## Setting Up Identity Provider
+
+1. Obtain SAML metadata from Sauce Labs Service Provider which is served under [this link](https://accounts.saucelabs.com/am/sso/metadata/https%3A%2F%2Faccounts.saucelabs.com%2Fsp).
```
Comment (aliciakrebs) [inline docs/basics/sso/setting-up-sso.md:29]:
```suggestion
1. Obtain SAML metadata from Sauce Labs Service Provider, which is served under [this link](https://accounts.saucelabs.com/am/sso/metadata/https%3A%2F%2Faccounts.saucelabs.com%2Fsp).
```

**Review by aliciakrebs — CHANGES_REQUESTED** [review]
I made some copy edits.

**Comment by przemek-sl** [comment]
@aliciakrebs I pushed fixes for all your comments.

**Review by aliciakrebs — APPROVED** [review]

**Pull request merged**